hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java

```diff
@@ -2876,10 +2876,17 @@ public Reader getConfResourceAsReader(String name) {
   protected synchronized Properties getProps() {
     if (properties == null) {
       properties = new Properties();
+      loadProps(properties, -1);
```

Comment:
I would really appreciate if you rewrite this so it doesn't use -1 as startIdx. A year from now I won't remember why we want to load from position -1.

Reply:
Good point @jojochuang .I will think about something. or maybe addd a comment.